extensions/amp-audio/1.0/test/test-amp-audio.js

```diff
@@ -0,0 +1,385 @@
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.getAttribute('preload')).to.be.equal('metadata');
+    });
+
+    it(
+      'should attach `<audio>` element and execute relevant actions for ' +
+        'layout="nodisplay"',
+      async () => {
+        element = await attachAndRun({
+          src: 'audio.mp3',
+          preload: 'none',
+          layout: 'nodisplay',
+        });
+
+        // Wait till rendering is finished
+        await waitForRender();
+
+        // Fetch audio element from shadowRoot
+        const {shadowRoot} = element;
+        const audio = shadowRoot.querySelector('audio');
+
+        expect(audio).to.not.be.null;
+
+        const impl = await element.getImpl();
+        impl.executeAction({method: 'play', satisfiesTrust: () => true});
+        expect(
+          impl.executeAction({method: 'isPlaying', satisfiesTrust: () => true})
+        ).to.be.true;
+
+        impl.executeAction({method: 'pause', satisfiesTrust: () => true});
+        expect(
+          impl.executeAction({method: 'isPlaying', satisfiesTrust: () => true})
+        ).to.be.false;
+      }
+    );
+
+    it('should load audio through sources', async () => {
+      element = attachAndRun(
+        {
+          width: 503,
+          height: 53,
+          autoplay: '',
+          preload: '',
+          muted: '',
+          loop: '',
+        },
+        [
+          {tag: 'source', src: 'audio.mp3', type: 'audio/mpeg'},
+          {tag: 'source', src: 'audio.ogg', type: 'audio/ogg'},
+          {tag: 'text', text: 'Unsupported.'},
+        ]
+      );
+
+      // Wait till rendering is finished
+      await waitForRender();
+
+      // Fetch audio element from shadowRoot
+      const {shadowRoot} = element;
+      const audio = shadowRoot.querySelector('audio');
+
+      expect(audio.tagName).to.equal('AUDIO');
+      expect(element.getAttribute('width')).to.be.equal('503');
+      expect(element.getAttribute('height')).to.be.equal('53');
+      expect(audio.offsetWidth).to.be.greaterThan(1);
+      expect(audio.offsetHeight).to.be.greaterThan(1);
+      expect(audio.hasAttribute('controls')).to.be.true;
+      expect(audio.hasAttribute('autoplay')).to.be.true;
+      expect(audio.hasAttribute('muted')).to.be.true;
+      expect(audio.hasAttribute('preload')).to.be.true;
+      expect(audio.hasAttribute('loop')).to.be.true;
+      expect(audio.hasAttribute('src')).to.be.false;
+      expect(audio.childNodes[0].tagName).to.equal('SOURCE');
+      expect(audio.childNodes[0].getAttribute('src')).to.equal('audio.mp3');
+      expect(audio.childNodes[1].tagName).to.equal('SOURCE');
+      expect(audio.childNodes[1].getAttribute('src')).to.equal('audio.ogg');
+      expect(audio.childNodes[2].nodeType).to.equal(Node.TEXT_NODE);
+      expect(audio.childNodes[2].textContent).to.equal('Unsupported.');
```

Comment:
This will not be defined once we have `sources` prop defined in `base-element.js` -- let's instead assert `expect(audio.childNodes).to.have.lengthOf(2)` since the third child is disregarded due to being not a `<source>` tag.